src/stan_math_backend/Stan_math_code_gen.ml

```diff
@@ -169,6 +169,22 @@ let pp_fun_def ppf Program.({fdrt; fdname; fdargs; fdbody; _}) =
     in
     pf ppf "%s(@[<hov>%a@]) " name (list ~sep:comma string) arg_strs
   in
+  let pp_sig_rs ppf name =
+    pp_template_decorator ppf templates ;
+    pp_returntype ppf fdargs fdrt ;
+    let first_three_args =
+      List.mapi ~f:(fun i x -> if i < 3 then x else "") args
+    in
+    let rest_args = List.mapi ~f:(fun i x -> if i >= 3 then x else "") args in
+    let arg_strs =
+      mk_extra_args prefix_extra_templates prefix_extra_args
+      @ List.filter ~f:(fun x -> x <> "") first_three_args
+      @ ["std::ostream* pstream__"]
+      @ List.filter ~f:(fun x -> x <> "") rest_args
```

Comment:
I am sure there is a nicer way to get the first three args and the rest as opposed to this foolish stuff. It works but not proud.

Reply:
I think `let first_three, rest = List.split_n args 3` should do it.

Reply:
Nice! Thanks a lot!